File: react/src/api/models/acroStack/services/dtos/openIddictManagement/OpenIddictApplicationDto.ts
/* oxlint-disable */

/**
 * @type object
 */
export type AcroStackServicesDtosOpenIddictManagementOpenIddictApplicationDto = {
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  id?: string;
  /**
   * @type string | undefined
   */
  clientId?: string | null;
  /**
   * @type string | undefined
   */
  displayName?: string | null;
  /**
   * @type string | undefined
   */
  clientType?: string | null;
  /**
   * @type string | undefined
   */
  consentType?: string | null;
  /**
   * @type string | undefined
   */
  clientSecret?: string | null;
  /**
   * @type array | undefined
   */
  permissions?: string[] | null;
  /**
   * @type array | undefined
   */
  redirectUris?: string[] | null;
  /**
   * @type array | undefined
   */
  postLogoutRedirectUris?: string[] | null;
  /**
   * @type array | undefined
   */
  requirements?: string[] | null;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  creationTime?: string;
};

File: AcroStack/Services/Dtos/OpenIddictManagement/OpenIddictApplicationDto.cs
using System;
using System.Collections.Generic;

namespace AcroStack.Services.Dtos.OpenIddictManagement;

public class OpenIddictApplicationDto
{
    public Guid Id { get; set; }
    public string? ClientId { get; set; }
    public string? DisplayName { get; set; }
    public string? ClientType { get; set; }
    public string? ConsentType { get; set; }
    public string? ClientSecret { get; set; }
    public List<string> Permissions { get; set; } = new();
    public List<string> RedirectUris { get; set; } = new();
    public List<string> PostLogoutRedirectUris { get; set; } = new();
    public List<string> Requirements { get; set; } = new();
    public DateTime CreationTime { get; set; }
}
